### PR TITLE
feat(getAccessor): correctly mock get accessors for mocked classes

### PR DESCRIPTION
**Pull request by Pmyl**
Thank you for your contribution to ts auto mock repo. 
Before submitting this PR, please make sure:

- [x] Your code builds clean without any errors or warnings
- [x] You have added tests

### CODE REVIEW

**Comment by Pmyl** [comment]
I have a question for you @uittorio , do I have to use the property cache? Since a getter is a property I think I should use it but I don't remember what that for

**Comment by uittorio** [comment]
That is correct! PropertyCache responsibility is to store the name of the property and consume it immediately after, it was a pain to pass it around

**Comment by Pmyl** [comment]
> That is correct! PropertyCache responsibility is to store the name of the property and consume it immediately after, it was a pain to pass it around

Just to understand the functionality better, why don't we use it to store method names as well?

**Comment by Pmyl** [comment]
Also, what test can I write to make sure a future me doesn't remove the cache usage?

**Comment by uittorio** [comment]
I need to double check but Methods and properties on classes and interfaces are using the propertyCache functionality. You can see it from the GetProperties function.

How do you test it?
the framewework test suite checks that the spies with jasmine have the correct names. You could extend those tests with getter and setter and is if that works :)

**Comment by uittorio** [comment]
Apparently it is also used when we generated the random string values.

File: src/transformer/descriptor/getAccessor/getAccessor.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import { GetFunctionReturnType } from '../method/functionReturnType';
+
+export function GetGetAccessorDeclarationDescriptor(
+  node: ts.GetAccessorDeclaration,
+  scope: Scope
+): ts.Expression {
+  const returnTypeNode: ts.Node = GetFunctionReturnType(node);
+  const returnType: ts.Expression = GetDescriptor(returnTypeNode, scope);
+
+  if (!node.name) {
```
Comment (uittorio) [inline src/transformer/descriptor/getAccessor/getAccessor.ts:13]: Why do we need this check?

**Review by uittorio — COMMENTED** [review]
Nice one!! Are you also doing the setter? :100:

**Comment by Pmyl** [comment]
@uittorio doing the setter may be a problem, what if we have both getter and setter? Is that going to break the functionality due to the fact that we're mocking the same property twice? I'll write a test to see if everything works fine

File: src/transformer/descriptor/getAccessor/getAccessor.ts (hunk 1 above)
Comment (Pmyl) [reply src/transformer/descriptor/getAccessor/getAccessor.ts]: removed this check

**Comment by Pmyl** [comment]
@uittorio I've done the changes to handle setters as well, in essence I removed the "not supported" warning by returning undefined and handled case when setter was defined before getter

**Review by uittorio — APPROVED** [review]

**Pull request merged**